src/handlers.rs

```diff
@@ -497,27 +501,35 @@ where
 
             log::debug!("Proposed psbt: {:#?}", psbt);
 
-
+    
             // basic flow for broadcasting: sign, extract tx, and then broadcast 
 
-            // Set sign options
-            let mut sign_options = SignOptions::default();
-            sign_options.trust_witness_utxo = true;
+            // Try trait input signer <----
+            dbg!(psbt.clone());
+            // Set sign options - just tryna fix an error here but trust witness utxo true is necessary
+            let mut sign_options1 = SignOptions::default();
+            sign_options1.trust_witness_utxo = true;
+            sign_options1.allow_all_sighashes = true;
+            sign_options1.remove_partial_sigs = false;
+
+            wallet.sign(&mut psbt, sign_options1.clone())?; // this returns a bool here
 
-            // Sign and finalize the PSBT
-            let finalized = wallet      // this returns a bool here
-                .sign(&mut psbt, sign_options);
+            // maybe try allow_grinding?
+            let finalized = wallet.finalize_psbt(&mut psbt, sign_options1)?; // this returns a bool here
```

Comment:
Can you paste the PSBT here after finalizing it?

Reply:
PartiallySignedTransaction {
    unsigned_tx: Transaction {
        version: 1,
        lock_time: PackedLockTime(
            300,
        ),
        input: [
            TxIn {
                previous_output: OutPoint {
                    txid: 0e0b096e7fcd290f777e8554c6af0424d53383e93ab9ca46a1ca02cbf4ee9b12,
                    vout: 0,
                },
                script_sig: Script(),
                sequence: Sequence(
                    4294967293,
                ),
                witness: Witness {
                    content: [],
                    witness_elements: 0,
                    last: 0,
                    second_to_last: 0,
                },
            },
            TxIn {
                previous_output: OutPoint {
                    txid: 574a168d4a2a7d9d20705fe50e65cbe3d91da12471d20c617ede209d798fac96,
                    vout: 0,
                },
                script_sig: Script(),
                sequence: Sequence(
                    4294967293,
                ),
                witness: Witness {
                    content: [],
                    witness_elements: 0,
                    last: 0,
                    second_to_last: 0,
                },
            },
        ],
        output: [
            TxOut {
                value: 2500010000,
                script_pubkey: Script(OP_0 OP_PUSHBYTES_20 f2703915adb4f455ccf874e2a5d39c3dfefcd341),
            },
            TxOut {
                value: 3999989791,
                script_pubkey: Script(OP_0 OP_PUSHBYTES_20 3e6e94664bd6704bb53c8e8192d2b0d38290ee51),
            },
        ],
    },
    version: 0,
    xpub: {},
    proprietary: {},
    unknown: {},
    inputs: [
        Input {
            non_witness_utxo: Some(
                Transaction {
                    version: 2,
                    lock_time: PackedLockTime(
                        0,
                    ),
                    input: [
                        TxIn {
                            previous_output: OutPoint {
                                txid: 0000000000000000000000000000000000000000000000000000000000000000,
                                vout: 4294967295,
                            },
                            script_sig: Script(OP_PUSHBYTES_2 c700 OP_0),
                            sequence: Sequence(
                                4294967295,
                            ),
                            witness: Witness {
                                content: [],
                                witness_elements: 0,
                                last: 0,
                                second_to_last: 0,
                            },
                        },
                    ],
                    output: [
                        TxOut {
                            value: 2500000000,
                            script_pubkey: Script(OP_0 OP_PUSHBYTES_20 18f52b5872979a41d0a2529f03f6cdf58ea6c1ca),
                        },
                        TxOut {
                            value: 0,
                            script_pubkey: Script(OP_RETURN OP_PUSHBYTES_36 aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf9),
                        },
                    ],
                },
            ),
            witness_utxo: Some(
                TxOut {
                    value: 2500000000,
                    script_pubkey: Script(OP_0 OP_PUSHBYTES_20 18f52b5872979a41d0a2529f03f6cdf58ea6c1ca),
                },
            ),
            partial_sigs: {},
            sighash_type: None,
            redeem_script: None,
            witness_script: None,
            bip32_derivation: {},
            final_script_sig: None,
            final_script_witness: Some(
                Witness {
                    content: [
                        71,
                        48,
                        68,
                        2,
                        32,
                        95,
                        51,
                        174,
                        187,
                        8,
                        12,
                        27,
                        30,
                        8,
                        254,
                        115,
                        54,
                        250,
                        131,
                        48,
                        196,
                        217,
                        217,
                        173,
                        80,
                        44,
                        172,
                        85,
                        208,
                        94,
                        139,
                        202,
                        21,
                        96,
                        66,
                        236,
                        156,
                        2,
                        32,
                        124,
                        23,
                        105,
                        105,
                        230,
                        177,
                        48,
                        42,
                        167,
                        195,
                        189,
                        149,
                        4,
                        94,
                        188,
                        197,
                        110,
                        213,
                        195,
                        224,
                        159,
                        64,
                        95,
                        218,
                        61,
                        44,
                        222,
                        103,
                        181,
                        250,
                        132,
                        203,
                        1,
                        33,
                        3,
                        222,
                        236,
                        250,
                        163,
                        16,
                        27,
                        104,
                        196,
                        225,
                        70,
                        85,
                        168,
                        173,
                        220,
                        118,
                        167,
                        209,
                        81,
                        41,
                        148,
                        239,
                        45,
                        139,
                        176,
                        135,
                        203,
                        116,
                        102,
                        8,
                        250,
                        88,
                        69,
                    ],
                    witness_elements: 2,
                    last: 72,
                    second_to_last: 0,
                },
            ),
            ripemd160_preimages: {},
            sha256_preimages: {},
            hash160_preimages: {},
            hash256_preimages: {},
            tap_key_sig: None,
            tap_script_sigs: {},
            tap_scripts: {},
            tap_key_origins: {},
            tap_internal_key: None,
            tap_merkle_root: None,
            proprietary: {},
            unknown: {},
        },
        Input {
            non_witness_utxo: None,
            witness_utxo: None,
            partial_sigs: {},
            sighash_type: None,
            redeem_script: None,
            witness_script: None,
            bip32_derivation: {},
            final_script_sig: None,
            final_script_witness: None,
            ripemd160_preimages: {},
            sha256_preimages: {},
            hash160_preimages: {},
            hash256_preimages: {},
            tap_key_sig: None,
            tap_script_sigs: {},
            tap_scripts: {},
            tap_key_origins: {},
            tap_internal_key: None,
            tap_merkle_root: None,
            proprietary: {},
            unknown: {},
        },
    ],
    outputs: [
        Output {
            redeem_script: None,
            witness_script: None,
            bip32_derivation: {},
            tap_internal_key: None,
            tap_tree: None,
            tap_key_origins: {},
            proprietary: {},
            unknown: {},
        },
        Output {
            redeem_script: None,
            witness_script: None,
            bip32_derivation: {},
            tap_internal_key: None,
            tap_tree: None,
            tap_key_origins: {},
            proprietary: {},
            unknown: {},
        },
    ],
}

Reply:
From the look of it, the PSBT looks malformed. It has only one witness in one of the inputs, and the other input is empty. There also don't seem to be any PSBT outputs, whereas there are two P2WPKH outputs in the unsigned transaction. 

I am not fully sure what's going wrong, but it seems like the payjoin party isn't returning the correct psbt. The one signature in the psbt is probably coming from the wallet itself. 

To dig deeper, you probably need to understand psbts a bit better. The easiest thing to do is go by the docs of the psbt structure and try to match why certain components could be missing. https://github.com/rust-bitcoin/rust-bitcoin/blob/83cf389a02d6147a4cd3fe9c42388c61abd40ec2/bitcoin/src/psbt/mod.rs#L44-L63

Reply:
The PSBT isn't malformed, it's partial! That info should be filled in by BDK-CLI. This receiver Will is tensting works with other payjoin implementations, the payjoin-cli sender and and even another BDK payjoin implementation ((bitmask)[https://github.com/diba-io/bitmask-core/pull/121]). I was able to replicate this bug. I'm thinking it might be an issue with the descriptor argument.

 @willowens14 next time you put logs in github you can fence (\`\`\` \`\`\`) it so it maintains formatting.

Reply:
I don't believe the PSBT is malformed, it's partial! That info should be filled in by BDK-CLI. This receiver Will is tensting works with other payjoin implementations, the payjoin-cli sender and and even another BDK payjoin implementation ((bitmask)[https://github.com/https://github.com/diba-io/bitmask-core/pull/121]). I was able to replicate this bug. I'm thinking it might be an issue with the descriptor argument.

@willowens14 next time you put logs in github you can fence (``` ```) it so it maintains formatting.

Reply:
Thanks for the detailed write-up here. That's a deep nuance I missed on BIP78/174 interaction.